# Linear Regression Model

## import libs

In [3]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
import numpy as np
import pandas as pd

## read data base

In [50]:
data_init = pd.read_csv('RESULTS_sobolsampling-2048.csv', index_col=1)
#print(f'{data_init}')
data_init = data_init.drop(data_init.columns[0], axis=1)
data_init = data_init.transpose()
print(f'{data_init}')

data = pd.read_csv('CLEANED_sobolsampling-2048.csv')
#print(f'{data}')
data = data.drop(data.columns[0], axis=1)
#data = data.transpose()
print(f'{data}')

#         SigC      SigH      EpsC      EpsH     density  density_err
1     0.329511  0.074231  1.036268  0.062599  891.797000        0.820
2     0.066754  0.316262  0.205940  0.119844    0.183657        0.009
3     0.193948  0.136800  0.751676  0.037376  664.747000        1.900
4     0.230588  0.235899  0.596203  0.100129  682.448000        0.380
5     0.242166  0.148891  0.406759  0.087550  688.667000        1.800
...        ...       ...       ...       ...         ...          ...
2044  0.242035  0.235637  0.735539  0.137124  708.349000        0.690
2045  0.230760  0.148096  1.005841  0.111508  714.162000        1.200
2046  0.194069  0.189609  0.357464  0.056020    0.000000      100.000
2047  0.066912  0.095994  0.782125  0.108488  651.054000        1.800
2048  0.329646  0.259110  0.444702  0.043977  707.088000        0.470

[2048 rows x 6 columns]
          SigC      SigH      EpsC      EpsH  density
0     0.329511  0.074231  1.036268  0.062599  891.797
1     0.193948  0.136800  0

## prepare data

In [52]:
X_init = data_init.drop('density', axis=1)
X_init = X_init.drop('density_err', axis=1)
print(f'{X_init}')

Y_init = data_init['density']
print(f'{Y_init}')

X_train_init, X_test_init, Y_train_init, Y_test_init = train_test_split(X_init, Y_init, test_size=0.05, random_state=29)

                                                    
X = data.drop('density', axis=1)
print(f'{X}')

Y = data['density']
print(f'{Y}')

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.05, random_state=29)

#         SigC      SigH      EpsC      EpsH
1     0.329511  0.074231  1.036268  0.062599
2     0.066754  0.316262  0.205940  0.119844
3     0.193948  0.136800  0.751676  0.037376
4     0.230588  0.235899  0.596203  0.100129
5     0.242166  0.148891  0.406759  0.087550
...        ...       ...       ...       ...
2044  0.242035  0.235637  0.735539  0.137124
2045  0.230760  0.148096  1.005841  0.111508
2046  0.194069  0.189609  0.357464  0.056020
2047  0.066912  0.095994  0.782125  0.108488
2048  0.329646  0.259110  0.444702  0.043977

[2048 rows x 4 columns]
1       891.797000
2         0.183657
3       664.747000
4       682.448000
5       688.667000
           ...    
2044    708.349000
2045    714.162000
2046      0.000000
2047    651.054000
2048    707.088000
Name: density, Length: 2048, dtype: float64
          SigC      SigH      EpsC      EpsH
0     0.329511  0.074231  1.036268  0.062599
1     0.193948  0.136800  0.751676  0.037376
2     0.230588  0.235899  0.596203  0.100129
3 

## create and train the model

In [54]:
## create the model
model_init = LinearRegression()

## train/fit the model
model_init.fit(X_train_init, Y_train_init)

## prediction using the test set
Y_prediction_init = model_init.predict(X_test_init)

## evaluate with Y_test
rmse_init = np.sqrt(mean_squared_error(Y_test_init, Y_prediction_init))
r2_init = r2_score(Y_test_init, Y_prediction_init)

print(f'RMSE_init: {rmse_init}')
print(f'R2_init: {r2_init}')
print(f'')

## create the model
model = LinearRegression()

## train/fit the model
model.fit(X_train, Y_train)

## Prediction using the test set
Y_prediction = model.predict(X_test)

## evaluate with Y_test
rmse = np.sqrt(mean_squared_error(Y_test, Y_prediction))
r2 = r2_score(Y_test, Y_prediction)

print(f'RMSE_cleaned: {rmse}')
print(f'R2_cleaned: {r2}')

RMSE_init: 107.03263221243395
R2_init: 0.2262922709240136

RMSE_cleaned: 70.98544157212672
R2_cleaned: 0.6015651228280929


## Result

- RMSE init: 107.03263221243395
- R2 init: 0.2262922709240136

and

- RMSE cleaned: 70.98544157212672
- R2 cleaned: 0.6015651228280929

genereally: 0 <= R2 <= 1, with R2 -> 1: better matching of Y_prediction to Y_test. 